cddlib: Update URLs, sha256

NessieCanCode opened the pull request:
Updated FTP URL for GitHub URL.

NessieCanCode committed:
Update package.py

Updated FTP URL for GitHub URL.

NessieCanCode committed:
Update package.py

NessieCanCode committed:
Update package.py

NessieCanCode committed:
Update package.py

NessieCanCode committed:
Update package.py

NessieCanCode committed:
Update package.py

eschnett commented on var/spack/repos/builtin/packages/cddlib/package.py:
```diff
@@ -13,16 +13,16 @@ class Cddlib(AutotoolsPackage):
+    homepage = "https://people.inf.ethz.ch/fukudak/cdd_home"
+    url      = "https://github.com/cddlib/cddlib/archive/refs/tags/0.94h.tar.gz"
+    maintainers = ['NessieCanCode']
+    version('0.94h', sha256='7382782c3834214b022c8b2898ed775a7bf915f2cb2acb73fa045d6fd9a3de33')
```
Newer versions are available. Since you're updating the hash anyway you could also add the newest version.

NessieCanCode replied:
It was a greedy move on my part, I only needed 0.94h, as the ftp server that was listed was timing out. but you are right, I can, so I've since added several of the version releases all the way to the latest version.

NessieCanCode committed:
Update package.py

NessieCanCode committed:
Update package.py

bernhardkaindl committed:
gmp@0.94h only, rm libtool dep, not all vers needed

I've ran into cddlib myself and had PR #26088 for it, closing mine.

Merging from my PR, with one further fix/improvement:

libgmp is only needed for 0.94
libtool not at all (also removed from my host, otherwise it uses that)

I also removed the span of versions between the original version
and the latest. Usually only the latest version is added as checksum,
but thanks to your url_from_version, all versions can be built even
automatically without question when using spack install -n to suppress
it asking if one really wants to install a version without a checksum.

In case you'd need a version in that span, you can submit a pull
request but as you wrote that 0.94h was enough for you, I'm assuming
that you it's ok for you to just have the latest version an 0.94h.

bernhardkaindl reviewed (approved):
Tested to build in a Ubuntu 18.04 container using it's gcc-8.4.0 without libtool and libgmp-dev
